ui/src/app/shared/common.ts

```diff
@@ -36,6 +39,30 @@ export const queryExtensionsDataTypes: Map<string, FieldDataType> = new Map([
   ['submission', FieldDataType.Date]
 ]);
 
+export const stringToTimeFrameMap: Map<String, TimeFrame> = new Map([
+  ['HOURS_1', TimeFrame.HOURS1],
```

Comment:
What is the string here? Why would it be different from the enum value which is already mapped for us from swagger codegen?

Reply:
How to use that to map a string to TimeFrame object?

Reply:
In [`dashboard.resolver.service.ts`](https://github.com/DataBiosphere/job-manager/pull/453/files#diff-573e628c01e7c1d012851d65411063b4R18):

```
timeFrame = (<any>TimeFrame)[route.queryParams['timeFrame']]
```

Reply:
When we encode the timeFrame in the url, the server find this parameter to be an unicode string, which dose not match to enum type in python. Should we do a conversion in javascript or python?

Reply:
Ah sorry what I posted was wrong, all you actually have to do is a type conversion in the javascript:

```
timeFrame = <TimeFrame>route.queryParams['timeFrame']
```

Reply:
It works! Thanks!